Guidelines/serviceOperations.md

```diff
@@ -0,0 +1,85 @@
+* Define stages and give them meaningful names ("Install Dependencies", "Build", "Test", "Deploy")
+
+## Kubernetes Service Definitions
+
+To have a microservice running on Kubernetes there is a need to have it's Kubernetes definitions (deployment and service yaml files) within the project folder.
+
+Proposal:
+
+* Kubernetes definitions should always be in a folder named `k8s` whithin the project folder
+* Use a consistent naming pattern: `deployment.yaml` and `service.yaml` for the definitions
+* Ingress configuration should be placed in a different repository (e.g. OIH-Infrastructure)
+* every Microservice need to have a checkable endpoint to have a liveliness and readiness check for the K8s
+
+
+### Why infrastructure repository
+
+Having a separate repository does provide advantages, for example – it could contain sidecar infrastructure like Wordpress or Jenkins and also scripts to do maintenance work or setup a new cluster.  
+The Ingress configuration should be placed here as well, because it is an abstraction of the load balancer configuration, which is related to the platform and not to a single service.  
+
+__-->In case of a monorepo, all the mentioned things could be placed in a own section within the monorepo.<--__
+
+
+# Monitoring
+
+is the most underrated part of operating a scaled platform. It could help finding sssues (peaks, errors, stack traces) as well as causing trouble (fill up diskspace caused by loops, resource consumption while analysing). 
+
+* Metrics
+* Logs
+* TCP/Http checks
+
+__For Metrics__: 
+*     The owner of the Microservice needs to define which metrics are necessary to keep and make some analytics on it
+
+__For Logs__:
+*     Should be done in JSON format to help log parser for analytics (Splunk, Logstash, Stackdriver). Log should be used for warning and errors so please watch the log level (debug only for local usage)
+
+__For TCP/HTTP__: 
+*     That type of checks are used to check healthyness of the service
+
+# Scaling
+
+There are different types for scaling a service. Using a cluster (horizontally and vertically) the approach is to have small services (microservices) which could be scaled out in starting more instances of the service. When deciding to go for huge service which needs a lot of resource the scaling method is vertical.
+
+For OHI we decided to use small services or microservices (there is a bunch of interpretation here).
+
+Some rules to take care of:
+
+* The service should be stateless so it could be moved any time from the cluster scheduler
+* if there is an error thrown which could not be catched somewhere the service needs to be restarted by K8S
+* keep an eye on the resources. Services with high RAM or CPU usage are not easy to spread in a cluster ENV.
+* The service should be able to run multiple times on a single node (if for a good reason that is not possible the cluster should care of the affinity)
+
```

Comment:
I think we should also define one more section: Reliability. 
General idea: In production environment every deployment should run in more then one instance.
This makes:
1. the whole platform won't fail in case if one deployment instances dies (e. g. just error in service, migration from one k8s node to another)
2. This make easier blue-green deployment strategy -- so no downtime during release. 

In case if it's hard to make service to run in parallel, there should be main instance and it's hot standby (e. g. use zookeeper/etcd/whatever to solve who is main and who is standby)
This requirement also may be relaxed if platform don't crash if one of services fails (e. g. some batch processing service. Dead service only introduce some additional delay)

Clearly that design solution from Scaling section often  solve problems described in Reliability section

Reply:
Sure you are correct. I don't mention that in the Document because i was the opinion that is default :) i will add that thx